openam-authentication/openam-auth-oauth2/src/main/java/org/forgerock/openam/authentication/modules/oauth2/OAuth.java

```diff
@@ -275,6 +286,9 @@ public int process(Callback[] callbacks, int state) throws LoginException {
                                 + "contained an unexpected value");
                         throw new AuthLoginException(BUNDLE_NAME, "incorrectState", null);
                     }
+                    if (!pkceCodeChallengeMethod.equalsIgnoreCase("none")) {
+                        pkceCodeVerifier = csrfStateToken.getValue(CoreTokenField.STRING_TWO);
```

Comment:
I think it's better to throw `AuthLoginException` if CSRF token doesn't contain `code_verifier`. Also, the debug log should be output.

Reply:
Thanks for adding exception handling. One more thing, need to add `notFoundCodeVerifier` to properties files.

Reply:
Thanks. I fixed it.

Reply:
Good, thanks.